cargo-scout-lib/src/scout/mod.rs

```diff
@@ -27,57 +28,55 @@ where
         }
     }
     pub fn run(&self) -> Result<Vec<Lint>, crate::error::Error> {
-        let diff_sections = self.vcs.sections(".")?;
-        let current_dir = std::fs::canonicalize(".")?;
+        let current_dir = std::fs::canonicalize(std::env::current_dir()?)?;
+        let diff_sections = self.vcs.sections(current_dir.clone())?;
         let mut lints = Vec::new();
-        let members = self.config.members();
+        let config_members = self.config.members();
+        let members = config_members.iter().map(|m| {
+            let mut member = current_dir.clone();
+            member.push(m);
+            member
+        });
         // There's no need to run the linter on members where no changes have been made
-        let relevant_members = members.iter().filter(|m| diff_in_member(m, &diff_sections));
+        let relevant_members = members.filter(|m| diff_in_member(m, &diff_sections));
         for m in relevant_members {
-            lints.extend(self.linter.lints(current_dir.join(m))?);
+            lints.extend(self.linter.lints(current_dir.clone().join(m))?);
         }
         println!("[Scout] - checking for intersections");
         Ok(lints_from_diff(&lints, &diff_sections))
     }
 }
 
-fn diff_in_member(member: &str, sections: &[Section]) -> bool {
-    for s in sections {
-        if s.file_name.starts_with(&member) {
-            return true;
+fn diff_in_member(member: &PathBuf, sections: &[Section]) -> bool {
+    if let Some(m) = member.to_str() {
+        for s in sections {
+            if s.file_name.starts_with(&m) {
+                return true;
+            }
         }
     }
     false
 }
 
-// Check if clippy_lint and git_section have overlapped lines
-fn lines_in_range(clippy_lint: &Span, git_section: &Section) -> bool {
-    // If git_section.line_start is included in the clippy_lint span
-    clippy_lint.line_start <= git_section.line_start && git_section.line_start <= clippy_lint.line_end ||
-    // If clippy_lint.line_start is included in the git_section span
-    git_section.line_start <= clippy_lint.line_start && clippy_lint.line_start <= git_section.line_end
+// Check if lint and git_section have overlapped lines
+fn lines_in_range(lint: &Lint, git_section: &Section) -> bool {
+    // If git_section.line_start is included in the lint span
+    lint.location.lines[0] <= git_section.line_start && git_section.line_start <= lint.location.lines[1] ||
+    // If lint.line_start is included in the git_section span
+    git_section.line_start <= lint.location.lines[0] && lint.location.lines[0] <= git_section.line_end
 }
 
-fn files_match(clippy_lint: &Span, git_section: &Section) -> bool {
+fn files_match(lint: &Lint, git_section: &Section) -> bool {
     // Git diff paths and clippy paths don't get along too well on Windows...
-    clippy_lint.file_name.replace("\\", "/") == git_section.file_name.replace("\\", "/")
+    lint.location.path.replace("\\", "/") == git_section.file_name.replace("\\", "/")
```

Comment:
I think there's a problem here: in the VCS module we return absolute paths, but the clippy lints use relative paths

Reply:
Oh that's weird, I thought I made everything use absolute paths from now on Oo, lemme have a look

Reply:
There's indeed a bug ! Nice catch !

I'm going to add a testcase to make sure it doesn't happen again

Reply:
Ok fixed it!